app/web/features/search/FilterDialog.tsx

```diff
@@ -252,8 +252,11 @@ export default function FilterDialog({
                 variant="standard"
                 id="num-guests-filter"
                 value={numberOfGuestFilter}
+                inputProps={{ min: 0 }}
                 onChange={(e) => {
-                  const tempNumOfGuest = parseInt(e.target.value);
+                  const convertedValue = parseInt(e.target.value);
+                  const tempNumOfGuest = !Number.isNaN(convertedValue) && convertedValue > 0 ? convertedValue : undefined;
+
```

Comment:
explanation:

- parseInt("") => NaN
- also, if user types -5 will cause an error
- aaand, if the type it's always number and NOT undefined, this input always contains a number, so isn't empty, which (IMHO) is annoying, plus when the user types a digit (ie: '1') result will be '01'.

with this solution I solve all the mentioned problems

Reply:
I think if you fix the types in the parent component you won't need so much logic, so let's see.

Reply:
but I need it to be a number, also being an string I can't compare if it's lower than 0 (or I think I shouldn't)

Reply:
Okay got it, I made a comment below for the other one, could you do the coercion for these others like that in a separate function variable like `handelNumGuestsChange`?

It will be more performant by preventing re-renders, especially since calculations are happening here.

Reply:
sure ✅